### PR TITLE
Backend Setup and AI Integration

### PR DESCRIPTION
**Pull request by MBHunter88**
This pull request sets up the foundational server infrastructure, with key features including API routes for user management, database integration, and OpenAI-powered conception plan generation. It also includes error handling and preparatory steps for OAuth integration as a future enhancement.

### Commits Summary:

Server Setup:
- Set up server architecture, including file structure and package dependencies.
- Created a test route in server.js to verify server functionality.

Routes and Database Connection:
- Established core routes for user management, including signup and login.
- Added a stretch goal for OAuth integration for enhanced security in the future.
- Connected the PostgreSQL database to manage user and plan data.

Error Messaging Enhancements:
- Improved error messages across all routes to provide better diagnostics during development and testing.

OpenAI API Integration:
- Integrated OpenAI API to generate a personalized conception plan based on user input.
- The generated plan is stored in the conception_plan database table.
- Personalized the AI-generated responses by using user profile information.

Comments for Clarity:
- Added comments to the codebase to provide clarity for future development, especially around TODO items for planned features.

### Outstanding Work/Stretch Goals:

OAuth integration for user authentication is outlined but not yet fully implemented.
Testing of AI routes with Postman has been partially conducted; more comprehensive tests are planned.

### Considerations:

Are there additional error handling measures or improvements needed?

### Testing Conducted:

Verified core routes (/signup, /login, /plan/generate/:id) via Postman.
Tested OpenAI API integration to ensure personalized plans are generated and stored successfully.

postman_login

postman_signup

postman_plan_generate

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+// TODO: Use the OAuth routes
+//app.use('/oauth', oauthRoutes);
+
+//POST /login - verify user info from db
+app.post('/login', async (req, res) => {
+    const { email, password } = req.body;
+    try {
+      const result = await db.query('SELECT * FROM users WHERE email = $1', [email]);
+      //TODO: If not using 0auth then hash password for direct db storage
```
Comment (Shikha2410) [inline server/server.js]: just curious- is hashing password feature also set as stretch goal?

**Comment by Shikha2410** [comment]
In PR Summary > Testing Conducted section, please attach screenshots of your verification effort on each core route via Postman. Screenshot should show the test input, whether the test was successful, and any other expected results in the test.

File: server/server.js
Comment (Shikha2410) [inline server/server.js:1]: Can we refactor this file into two to categorize user management routes in one and conception plan generation routes in another file?

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                    Method choice: ${method_choice}, 
+                    Donor preference: ${donor_preference}, 
+                    Known fertility issues: ${known_fertility_issues}, 
+                    Timeline: ${timeline}.`,
+          },
+      ],
+  });
+
+  const aiResponse = completion.choices[0].message.content;
+  console.log('AI Response:', aiResponse);
```
Comment (Shikha2410) [inline server/server.js]: Can comment out and add TODO to eventually remove debug line.

**Review by Shikha2410 — COMMENTED** [review]
In PR Summary > Testing Conducted section, please attach screenshots of your verification effort on each core route via Postman. Screenshot should show the test input, whether the test was successful, and any other expected results in the test.

Left other minor feedback. Will approve after you've addressed the same.

**Comment by jsnorek** [comment]
This looks great! Very detailed set up. Excited to see the OAuth progress

File: server/server.js (hunk 1 above)
Comment (MBHunter88) [reply server/server.js]: For now, yes it is a stretch goal, I want to prioritize the oAuth stretch goal and use password hashing as a backup plan.

File: server/server.js
Comment (MBHunter88) [reply server/server.js:1]: I have successfully refactored the server.js file into two separate route files:

userRoutes.js: Handles all user-related routes such as signup, login, and profile updates.
planRoutes.js: Manages the generation of conception plans using AI.

Thank you for suggesting the refactor—it significantly improves the code structure!

**Review by Shikha2410 — APPROVED** [review]
Good to go!

**Pull request merged**